dwave/cloud/solver.py

```diff
@@ -408,16 +499,93 @@ def upload_bqm(self, bqm):
         Note:
             To use this method, dimod package has to be installed.
         """
+        return self.upload_problem(bqm)
+
+
+class DQMSolver(BaseUnstructuredSolver):
+    """Class for D-Wave unstructured discrete quadratic model solvers.
+
+    This class provides a :term:`DQM` sampling
+    method and encapsulates the solver description returned from the D-Wave
+    cloud API.
 
+    Args:
+        client (:class:`~dwave.cloud.client.Client`):
+            Client that manages access to this solver.
+
+        data (`dict`):
+            Data from the server describing this solver.
+
+    Note:
+        Events are not yet dispatched from unstructured solvers.
+    """
+
+    _handled_problem_types = {"dqm"}
+    _handled_encoding_formats = {"bq"}
+
+    def _encode_problem_for_upload(self, dqm):
         try:
-            data = bqm_as_file(bqm)
+            # note: SpooledTemporaryFile currently returned by DQM.to_file
+            # does not implement io.BaseIO interface, so we use the underlying
+            # (and internal) file-like object for now
+            data = dqm.to_file()._file
         except Exception as e:
-            logger.debug("BQM conversion to file failed with %r, "
+            logger.debug("DQM conversion to file failed with %r, "
                          "assuming data already encoded.", e)
-            # assume `bqm` given as file, ready for upload
-            data = bqm
+            # assume `dqm` given as file, ready for upload
+            data = dqm
 
-        return self.client.upload_problem_encoded(data)
+        logger.debug("Problem encoded for upload: %r", data)
+        return data
+
+    def sample_bqm(self, bqm, **params):
```

Comment:
Does this need to be "public"? Just not necessarily something that we want to encourage folks to do

Reply:
Not sure what you mean by "this". `sample_bqm` method?

Reply:
Oh, you mean on the `DQMSolver`. Well, to make the new type better fit in, for now, yes.

Reply:
I wonder if maybe a doc string that says "it's better to use a BQM solver" or similar would be good.

Reply:
Btw, I would not document this method, that's for sure. Like I wouldn't *many* other "public" methods in this module. /cc @JoelPasvolsky